### ISSUE TITLE
Missing passing function as arguments

### ISSUE THREAD

**Issue by InDIOS** [NectarJS/nectarjs#111]
Hi again @adrien-thierry, there is no way to pass a `function` as argument, instead of store it in a variable first and then pass it as argument?
Example:
```js
function some(cb) {
  cb();
}
some(function () {
  console.log('Callback');
});
```
This code cause an error in outputted C++ code:
```c
// stuff ...
some(function () {
// -------^ here is the error
  console["log"]("Inside a callback");
});
// more code ...
```
I've analyzed the `createAnon` function inside `compiler/native/lib` and you only process `FunctionExpression` as init value of  `VariableDeclarator`s, leaving other cases like `CallExpression` arguments as is.

**Comment by adrien-thierry** [comment]
Hi again,

Thank you, i will correct this behabior

**Comment by InDIOS** [comment]
Have you tought about using another stratagy to generate C++ code?  l've seen that is a bit difficult to change function expression to an equivalent C++ code with lambdas because using regular expression may cause error for example, when a string contain some javascript expression. I have an idea, you can use babel to make all changes to js code and use a custom js code generator to write C++. I mean that, instead to write js code with babel, you can, for example, use this generator to get a FunctionExpression node and write a C++ lambda. Sorry if i'm not so clear. If you have any doubt, you can ask me whatever you want.

**Comment by adrien-thierry** [comment]
Hi,

In fact, i don't want NectarJS to be too much dependant of Babel because maybe I will remove it in a near future to be able to self compile NectarJS.

I will apply another solution for now: I will create a temporary var just before the function call, and replace the lambda by this temp var

**Comment by adrien-thierry** [comment]
Behaviour corrected in a8a769b

Thank you for your help !
